doc/Makefile

```diff
@@ -33,39 +33,41 @@ help:
 	@echo "   specify RELEASE=name to publish as a tagged release version"
 	@echo "   and placed in a version subfolder.  Requires repo merge permission."
 
-.PHONY: help Makefile
+.PHONY: help Makefile copy-to-sourcedir doxy content kconfig html singlehtml clean publish
 
 # Generate the doxygen xml (for Sphinx) and copy the doxygen html to the
 # api folder for publishing along with the Sphinx-generated API docs.
 
-doxy:
-	$(Q)(cat acrn.doxyfile) | doxygen - > doc.log 2>&1
-	$(Q)find doxygen/xml/* | xargs sed -i 's/simplesect kind="pre"/simplesect kind="preconditions"/'
+copy-to-sourcedir:
+	$(Q)mkdir -p $(SOURCEDIR)
+	$(Q)rsync -rt --exclude=$(BUILDDIR) . $(SOURCEDIR)
 
-content:
-	$(Q)scripts/extract_content.py . misc
+doxy: copy-to-sourcedir
+	$(Q)(cat acrn.doxyfile ; echo "OUTPUT_DIRECTORY=$(SOURCEDIR)/doxygen" ) | doxygen - > $(BUILDDIR)/doc.log 2>&1
+	$(Q)find $(SOURCEDIR)/doxygen/xml/* | xargs sed -i 's/simplesect kind="pre"/simplesect kind="preconditions"/'
 
-kconfig:
+content: copy-to-sourcedir
+	$(Q)scripts/extract_content.py $(SOURCEDIR) misc
+
+kconfig: copy-to-sourcedir
 	$(Q)srctree=../hypervisor \
-	    python3 scripts/genrest.py Kconfig reference/kconfig/
+	    python3 scripts/genrest.py Kconfig $(SOURCEDIR)/reference/kconfig/
 
 pullsource:
 	$(Q)scripts/pullsource.sh
 
-html: doxy content kconfig
-	-$(Q)$(SPHINXBUILD) -t $(DOC_TAG) -b html -d $(BUILDDIR)/doctrees $(SOURCEDIR) $(BUILDDIR)/html $(SPHINXOPTS) $(OPTS) >> doc.log 2>&1
-	$(Q)./scripts/filter-doc-log.sh doc.log
+html: copy-to-sourcedir doxy content kconfig
+	-$(Q)$(SPHINXBUILD) -t $(DOC_TAG) -b html -d $(BUILDDIR)/doctrees $(SOURCEDIR) $(BUILDDIR)/html $(SPHINXOPTS) $(OPTS) >> $(BUILDDIR)/doc.log 2>&1
+	$(Q)./scripts/filter-doc-log.sh $(BUILDDIR)/doc.log
 
 singlehtml: doxy content kconfig
-	-$(Q)$(SPHINXBUILD) -t $(DOC_TAG) -b singlehtml -d $(BUILDDIR)/doctrees $(SOURCEDIR) $(BUILDDIR)/html $(SPHINXOPTS) $(OPTS) >> doc.log 2>&1
-	$(Q)./scripts/filter-doc-log.sh doc.log
+	-$(Q)$(SPHINXBUILD) -t $(DOC_TAG) -b singlehtml -d $(BUILDDIR)/doctrees $(SOURCEDIR) $(BUILDDIR)/html $(SPHINXOPTS) $(OPTS) >> $(BUILDDIR)/doc.log 2>&1
+	$(Q)./scripts/filter-doc-log.sh $(BUILDDIR)/doc.log
 
 # Remove generated content (Sphinx and doxygen)
 
 clean:
-	rm -fr $(BUILDDIR) doxygen
-	rm -fr misc
-	rm -fr reference/kconfig/*.rst
+	rm -fr $(BUILDDIR)
```

Comment:
You should also keep the removal of these legacy folders for a while to clean up the "contamination" of the source folder done by the previous Makefile, and because you're also updating the .gitignore file to not ignore these folders.  Put back:
```
        rm -fr misc
	rm -fr reference/kconfig/*.rst
```

Reply:
My thinking was that removing those would "force" the use to look at the state of his tree (and now that they are removed from `.gitignore` they would show up in a `git status` call). But your approach is better in fact, I've put them back.